gpii/node_modules/processReporter/src/ProcessReporter.js

```diff
@@ -0,0 +1,224 @@
+        }
+    }, function (err) {
+        request.events.onError.fire({
+            statusCode: 500,
+            message: "Failed to read processReporter source. " + (err.message || err.error),
+            isError: true
+        });
+    });
+};
+
+fluid.defaults("gpii.processReporter.live", {
+    gradeNames: ["gpii.processReporter.base"],
+    invokers: {
+        get: {
+            funcName: "gpii.processReporter.live.get",
+            args: ["{arguments}.0", "{processReporter}", "{flowManager}.solutionsRegistryDataSource"]
+        },
+        getProcess: {
+            funcName: "gpii.processReporter.live.getProcess",
+            args: ["{arguments}.0", "{processReporter}", "{flowManager}.solutionsRegistryDataSource"]
+        }
+    }
+});
+
+gpii.processReporter.live.get = function (request, processReporter, solutionsRegistryDataSource) {
+    var runningSolutions = [];
+    solutionsRegistryDataSource.get({os: processReporter.platformReporter.reportPlatform().id}, function onSuccess(entries) {
+        fluid.each(entries, function (entry, entryId) {
+            if (!runningSolutions.some(function (s) { return s.id === entryId; })) {
+                var foundEntry = fluid.find(entry.isRunning, function (runningSolutionsReporter) {
+                    var resolvedName = processReporter.nameResolver.resolveName(runningSolutionsReporter.type, "processReporter");
+                    var isRunning = fluid.invokeGradedFunction(resolvedName, runningSolutionsReporter);
+                    return { id: entryId, running: isRunning };
+                }, null);
+                if  (foundEntry !== null) {
+                    runningSolutions.push(foundEntry);
+                }
+            }
+        });
+        processReporter.fireResponse(request, runningSolutions);
+    });
+};
+
+gpii.processReporter.live.getProcess = function (request, processReporter, solutionsRegistryDataSource) {
+    var solutionID = request.req.params.solutionID;
+    var result;
+    solutionsRegistryDataSource.get({os: processReporter.platformReporter.reportPlatform().id}, function onSuccess(entries) {
+        var solution = fluid.find(entries, function (anEntry, entryID) {
+            if (entryID === solutionID) {
+                return anEntry;
+            }
+        }, null);
+        if (solution === null) {
+            result = { id: solutionID, running: false };
+        }
+        else {
+            result = fluid.find(solution.isRunning, function (runningSolutionsReporter) {
+                var resolvedName = processReporter.nameResolver.resolveName(runningSolutionsReporter.type, "processReporter");
+                var isRunning = fluid.invokeGradedFunction(resolvedName, runningSolutionsReporter);
+                return { id: solutionID, running: isRunning };
+            }, {id: solutionID, running: false});
+        }
+        processReporter.fireResponse(request, result);
+    });
+};
+
+fluid.defaults("gpii.platformReporter.native", {
```

Comment:
Removed.  But, I wonder if this platform information is necessary.  Does the client need to know the platform?